### PR TITLE
Basic routing

### PR DESCRIPTION
**Pull request by scottyplunkett**
- Removes Main Class
- Adds HTTP Server, HTTP Response, HTTP Request, and Router Classes
- Adds Pages directory and Static HTML Pages
- Breaks up Original Server Components
- Serves Basic GET Request for Static Pages

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,8 +5,27 @@
...
+                    <source>1.8</source>
+                    <target>1.8</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
```
Comment (esmevane) [inline pom.xml:25]: What is jUnit Jupiter?

File: src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPRequest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.scottyplunkett.server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+class HTTPRequest {
```
Comment (esmevane) [inline src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPRequest.java:7]: Is this the object you were asking about `static` methods on?

Check out this pseudocode version of this file:

```java
class HTTPRequest {
  private String s;
  ArrayList<String> request = new ArrayList<>();
  
  ArrayList<String> read(BufferedReader in) throws IOException {
    while ((s = in.readLine()) != null) {
      request.add(s);
      System.out.println(s);
      if (s.isEmpty()) {
        break;
      }
    }
    return request;
  }
  
  static ArrayList<String> read(BufferedReader in) throws IOException {
    HTTPRequest singleRequest = new HTTPRequest();
    
    return singleRequest.read(in);
  }
}
```

So, here you've got basically the exact same thing, but the interface is a little different.  The static method is invoking an instance method on itself.  This basically broadens the surface area of your object.  Now you can do things like assert against values, create instances at different states in the lifecycle, and some other tricks.

For example:

* Making sure we start with a clean state, just assert that request and / or s are empty to begin with.
* To make sure it builds from the reader correctly, instantiate an object and call read, then assert that request and / or s are no longer empty.

You can also control error handling, and avoid consuming streams too early, etc.

File: src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    static String getDate() {
+        Calendar calendar = Calendar.getInstance();
+        String format = "EEE, dd MMM yyyy HH:mm:ss z";
+        SimpleDateFormat dateFormat = new SimpleDateFormat(format, Locale.US);
+        dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"));
+        return dateFormat.format(calendar.getTime());
+    }
+
+    static String buildHeaders(String responseCode,
```
Comment (esmevane) [inline src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java:29]: I see `buildHeaders` and I think of a `Headers` object which can do a `build` operation.  There's also kind of a large amount of arguments here, which is another symptom of a separate object (several different pieces of state to manage).

File: src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        String contentTypeLine = "Content-Type: " + contentType + CRLF;
+        return  statusLine +
+                dateLine +
+                contentTypeLine;
+
+    }
+
+    private static String getResponseBodyContent(Path filePath) throws IOException {
+        String stringFromFile = Files.lines( filePath ).collect( Collectors.joining() );
+        System.out.print(stringFromFile);
```
Comment (esmevane) [inline src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java:46]: Maybe a good refactor pass here would be to take all of these `System.out.print` statements and instead hand around a `Logger` which can manage these outputs.

File: src/com/scottyplunkett/server/com/scottyplunkett/server/Router.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.scottyplunkett.server;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Router {
+        public static Path route(String requestLine) {
+            if ("GET / HTTP/1.1".equals(requestLine))
```
Comment (esmevane) [inline src/com/scottyplunkett/server/com/scottyplunkett/server/Router.java:8]: If you remember our talk about conditionals - _this_ would be a great example of a conditional which could be refactored into sort of a strategy matcher / strategy handler pattern.

**Review by esmevane — APPROVED** [review]
This looks pretty good, actually!  I don't want to hold it up to request changes, but I _do_ want you to think about refactors and different approaches you could go when you have some time later.

File: src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPRequest.java (hunk 2 above)
Comment (scottyplunkett) [reply src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPRequest.java:7]: Thank you! This sets me on the right path for refactoring toward data boundaries and testing values

File: src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java (hunk 3 above)
Comment (scottyplunkett) [reply src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java:29]: 100% agree, header logic and construction is next on my list

File: src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java (hunk 4 above)
Comment (scottyplunkett) [reply src/com/scottyplunkett/server/com/scottyplunkett/server/HTTPResponse.java:46]: 👍 I already removed these in the next PR and will add a Trello card for logging

File: src/com/scottyplunkett/server/com/scottyplunkett/server/Router.java (hunk 5 above)
Comment (scottyplunkett) [reply src/com/scottyplunkett/server/com/scottyplunkett/server/Router.java:8]: Thanks for giving me direction on this, I wasn't quite sure what I wanted to do here but hate its ugliness

**Pull request merged**